core/Url.php

```diff
@@ -829,4 +829,59 @@ public static function getHostFromServerNameVar()
         }
         return $host;
     }
+
+    /**
+     * Add campaign parameters to URLs linking to matomo.org to improve understanding of how online help is being
+     * used for different parts of the application, no personally identifiable information is included, just the area
+     * of the application from which the link originated.
+     *
+     * @param string|null $url      eg. www.matomo.org/faq/123 or https://matomo.org/faq/456
+     * @param string|null $campaign Optional campaign override, defaults to 'Matomo_App'
+     * @param string|null $source   Optional campaign source override, defaults to either 'Matomo_App_OnPremise' or
+     *                              'Matomo_App_Cloud'
+     * @param string|null $medium   Optional campaign medium, defaults to [module].[action] where module and action are
+     *                              taken from the currently viewed application page, eg. 'CoreAdminHome.trackingCodeGenerator'
+     *
+     * @return string|null      www.matomo.org/faq/123?mtm_campaign=Matomo_App&mtm_source=Matomo_App_OnPremise&mtm_medium=CoreAdminHome.trackingCodeGenerator
+     */
+    public static function addCampaignParametersToMatomoLink(?string $url = null, ?string $campaign = null,
```

Comment:
As you seem to use that method in all plugins as well, it will be required to update the version in `Version.php` and the minimum required version in all plugins that use this new method.